model/configuration.py

```diff
@@ -236,6 +257,45 @@ def cache_key(self):
         # bit and by_cache_key() not as much.
         return self.id
 
+    @classmethod
+    def get_integrations_of_goal(cls, _db, goal):
+        """Return all external integrations by goal type.
+        """
+        integrations = _db.query(cls).filter(
+            cls.goal==goal
+        ).order_by(
+            cls.name
+        ).all()
```

Comment:
You don't need to call `all()` here -- the query wil lbe run when the results are needed. This will allow someone to call `ExternalIntegration.for_goal` and then tack on any extra query conditions they need.